src/condor_utils/condor_scitokens.cpp

```diff
@@ -328,12 +341,34 @@ htcondor::validate_scitoken(const std::string &scitoken_str, std::string &issuer
 
 		char *scopes_char = nullptr;
 		if (!(*scitoken_get_claim_string_ptr)(token, "scope", &scopes_char, nullptr)) {
+			bool compute_create = false;
+			bool compute_modify = false;
+			bool compute_cancel = false;
 			StringList scopes_list(scopes_char);
 			scopes_list.rewind();
 			free(scopes_char);
 			char *scope;
 			while ( (scope = scopes_list.next()) ) {
 				scopes.emplace_back(scope);
+				if (foreign_token) {
+					// With a foreign token, the ACLs above will be empty,
+					// so we need to search the scopes here for HTCondor
+					// authorization levels.
+					if (strncmp(scope, "condor:/", 8) == 0) {
+						authz.emplace_back(&scope[8]);
```

Comment:
what if &scope[8] is the empty string?

Reply:
The text after the '/' is expected to be the name of a cedar auth level. If it's the empty string, then that's what's added to the list auth levels. It should be treated like any other invalid auth level name (i.e. ignored). The preceding loop over acls behaves the same way (will add an empty string).